### PR TITLE
Added BaseGroup::findAllChildrenOfType utility method

### PR DESCRIPTION
**Pull request by mmarineBlueQuartz**
* Added to DataStructure unit tests

### CODE REVIEW

**Review by imikejackson — APPROVED** [review]
Rebase on develop and merge

File: src/complex/DataStructure/BaseGroup.hpp [hunk 1]
```diff
@@ -142,6 +143,36 @@ class COMPLEX_EXPORT BaseGroup : public DataObject
...
+  std::set<std::shared_ptr<T>> findAllChildrenOfType(bool recursive = false) const
+  {
+    std::set<std::shared_ptr<T>> foundItems;
+    for(const auto& [key, ptr] : *this)
+    {
+      if(auto typePtr = std::dynamic_pointer_cast<T>(ptr))
+      {
+        foundItems.insert(typePtr);
+      }
+      if(recursive == true)
```
Comment (JDuffeyBQ) [inline src/complex/DataStructure/BaseGroup.hpp:164]:
```suggestion
      if(recursive)
```

File: src/complex/DataStructure/BaseGroup.hpp [hunk 2]
```diff
@@ -142,6 +143,36 @@ class COMPLEX_EXPORT BaseGroup : public DataObject
...
+   * @param recursive = false
+   * @return std::vector<std::shared_ptr<T>>
+   */
+  template <class T>
+  std::set<std::shared_ptr<T>> findAllChildrenOfType(bool recursive = false) const
+  {
+    std::set<std::shared_ptr<T>> foundItems;
+    for(const auto& [key, ptr] : *this)
+    {
+      if(auto typePtr = std::dynamic_pointer_cast<T>(ptr))
```
Comment (JDuffeyBQ) [inline src/complex/DataStructure/BaseGroup.hpp:160]:
```suggestion
      if(auto typePtr = std::dynamic_pointer_cast<T>(ptr); typePtr != nullptr)
```

File: src/complex/DataStructure/BaseGroup.hpp [hunk 3]
```diff
@@ -142,6 +143,36 @@ class COMPLEX_EXPORT BaseGroup : public DataObject
...
+      {
+        foundItems.insert(typePtr);
+      }
+      if(recursive == true)
+      {
+        if(auto groupPtr = std::dynamic_pointer_cast<BaseGroup>(ptr))
```
Comment (JDuffeyBQ) [inline src/complex/DataStructure/BaseGroup.hpp:166]:
```suggestion
        if(auto groupPtr = std::dynamic_pointer_cast<BaseGroup>(ptr); groupPtr != nullptr)
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

**Review by JDuffeyBQ — APPROVED** [review]

**Pull request merged**